### PR TITLE
Combine similar behaviour of view controllers

### PR DESCRIPTION
**Pull request by IvanJanuary**

### CODE REVIEW

File: NetworkRequest/NetworkRequest.xcodeproj/project.xcworkspace/xcuserdata/ivan.xcuserdatad/UserInterfaceState.xcuserstate
Comment (Lana-Resquit) [inline NetworkRequest/NetworkRequest.xcodeproj/project.xcworkspace/xcuserdata/ivan.xcuserdatad/UserInterfaceState.xcuserstate:1]: This file should not be here

File: NetworkRequest/NetworkRequest/Controller/RepositoryViewController.swift [hunk 1]
```diff
@@ -35,28 +35,11 @@ class RepositoryViewController: UIViewController, UISearchBarDelegate {
...
-                let searchResult = try decoder.decode(T.self, from: data) // T.self instead RepositorySearchResult.self
-                DispatchQueue.main.async {
-                    self.repositories = searchResult.items as? [Repository] ?? [] // unwrapped as? [Repository] ?? []
-                    self.tableView.reloadData()
-                }
-            } catch {
-                print("Decoding error JSON: \(error)")
-            }
+        let helper = GitHubHelper()
+        helper.search(withQuery: api, type: RepositorySearchResult.self) { searchItems in
```
Comment (Lana-Resquit) [inline NetworkRequest/NetworkRequest/Controller/RepositoryViewController.swift]: Read something about [weak self] in closures

File: NetworkRequest/NetworkRequest.xcodeproj/project.xcworkspace/xcuserdata/ivan.xcuserdatad/UserInterfaceState.xcuserstate
Comment (IvanJanuary) [reply NetworkRequest/NetworkRequest.xcodeproj/project.xcworkspace/xcuserdata/ivan.xcuserdatad/UserInterfaceState.xcuserstate:1]: yeah, it shouldn't be here, but it's starting to create again

Comment (IvanJanuary) [reply NetworkRequest/NetworkRequest.xcodeproj/project.xcworkspace/xcuserdata/ivan.xcuserdatad/UserInterfaceState.xcuserstate:1]: Fixed

**Review by Lana-Resquit — APPROVED** [review]

**Pull request merged**